internal/clients/aws.go

```diff
@@ -111,6 +113,37 @@ func (m *metaOnlyPrimary) Meta() any {
 	return m.meta
 }
 
+// withExternalAPICallCounter configures an AWS SDK v2 stack (client)
+// with an API call counter.
+func withExternalAPICallCounter(stack *middleware.Stack) error {
```

Comment:
Maybe it will help the future maintainers to explain how this middleware works, i.e., AWS Go SDK v2 makes the service name and the operation name available in the context and we install a middleware to extract them from the context and increment the upjet metric with the appropriate label values.